kms-connector/crates/tx-sender/tests/integration_tests.rs

```diff
@@ -18,148 +23,158 @@ use tx_sender::core::{
     DbKmsResponsePicker, DbKmsResponseRemover, TransactionSender, tx_sender::TransactionSenderInner,
 };
 
+#[dtor]
```

Comment:
My concern with using `dtor` (albeit it being super useful on a high-level) is that the behavior is not well-define, it is platform-specific and has races. For example, I tried to use it for test cleanup in the coprocessor and the issue was that thread-local storage was no longer available in the dtor and sometimes it worked, sometimes not.

See the Warnings section here: https://crates.io/crates/ctor

I don't know of a better solution to be honest, but to run tests one at a time and constructing/destructing the test instance in a well-defined manner in each test.